Mathlib/Data/Multiset/Pi.lean

```diff
@@ -16,74 +16,97 @@ namespace Multiset
 
 section Pi
 
-variable {α : Type*}
-
 open Function
 
-/-- Given `δ : α → Type*`, `Pi.empty δ` is the trivial dependent function out of the empty
+namespace Pi
+variable {ι : Type*} [DecidableEq ι] {α : ι → Sort*}
+
+/-- Given `α : ι → Type*`, `Pi.empty α` is the trivial dependent function out of the empty
 multiset. -/
-def Pi.empty (δ : α → Sort*) : ∀ a ∈ (0 : Multiset α), δ a :=
+def empty {ι : Type*} (α : ι → Sort*) : ∀ a ∈ (0 : Multiset ι), α a :=
   fun.
 #align multiset.pi.empty Multiset.Pi.empty
 
-universe u v
-variable [DecidableEq α] {β : α → Type u} {δ : α → Sort v}
+variable {i : ι} {m : Multiset ι}
 
-/-- Given `δ : α → Type*`, a multiset `m` and a term `a`, as well as a term `b : δ a` and a
-function `f` such that `f a' : δ a'` for all `a'` in `m`, `Pi.cons m a b f` is a function `g` such
-that `g a'' : δ a''` for all `a''` in `a ::ₘ m`. -/
-def Pi.cons (m : Multiset α) (a : α) (b : δ a) (f : ∀ a ∈ m, δ a) : ∀ a' ∈ a ::ₘ m, δ a' :=
-  fun a' ha' => if h : a' = a then Eq.ndrec b h.symm else f a' <| (mem_cons.1 ha').resolve_left h
+/-- Given `α : ι → Sort*`, a multiset `m` and a term `i`, as well as a term `a : α i` and a
+function `f` such that `f j : α j` for all `j` in `m`, `Pi.cons a f` is a function `g` such
+that `g k : α k` for all `k` in `i ::ₘ m`. -/
+def cons (a : α i) (f : ∀ j ∈ m, α j) : ∀ j ∈ (i ::ₘ m), α j :=
```

Comment:
Are you sure it's a good idea to make `i` and `m` implicit here? What's your reasoning?